File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) 
    {
     int low=0,med=0;
     int high=nums.size()-1;
     
        while(med<=high)
        {
            switch(nums[med])
            {
                case 0: swap(nums[low++],nums[med++]);
                    break;
                case 1:
                    med++;
                    break;
                case 2:
                    swap(nums[med],nums[high--]);
                    break;
                    
            }
        }
         
    }
};

File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        
        int c0=0,c1=0,c2=0;
        for(int i=0;i<nums.length;++i)
        {
            if(nums[i]==0)
                 c0++;
            else if(nums[i]==1)
                 c1++;
            else if(nums[i]==2)
                 c2++;
        }
        for(int i=0;i<nums.length;++i)
        {
            if(i<c0)
                nums[i]=0;
            else if(i>=c0&&i<c0+c1)
                nums[i]=1;
            else if(i>=c1&&i<c0+c1+c2)
                nums[i]=2;
        }
    }
}